pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -189,6 +191,10 @@ func (r *shootMaintenanceReconciler) reconcile(ctx context.Context, shoot *garde
 		shoot.Spec.Kubernetes.Version = *updatedKubernetesVersion
 	}
 
+	if hasMaintainNowAnnotation(shoot) {
+		delete(shoot.Annotations, v1beta1constants.GardenerOperation)
+	}
```

Comment:
why was this added / why is this needed?

Reply:
I think this is needed for the case when RotateSSHKeypairOnMaintenance is disabled and the Shoot lastOperation is not in Succeeded state. Currently, gardener.cloud/operation=reconcile annotation is added only when the Shoot lastOperation is Succeeded. It is actually something that was removed by https://github.com/gardener/gardener/pull/4224.